src/python/pants/core/goals/export.py

```diff
@@ -156,8 +155,6 @@ async def export(
     all_results = await MultiGet(Get(ExportResults, ExportRequest, request) for request in requests)
     flattened_results = [res for results in all_results for res in results]
 
-    await _warn_on_non_local_environments(targets, "the `export` goal")
```

Comment:
this is dead because above we throw if there are any `targets`, right?

Reply:
Yup! In effect this check never ran usefully, it just got to check targets before the python backend errored out. It became more apparent when I lifted the error to the goal instead of the union rule.